add_new_dc_test.py

```diff
@@ -0,0 +1,120 @@
+import warnings
+from typing import Tuple, List
+
+from cassandra import ConsistencyLevel
+from cassandra.query import SimpleStatement  # pylint: disable=no-name-in-module
+
+from longevity_test import LongevityTest
+from sdcm.cluster import BaseNode
+from sdcm.stress_thread import CassandraStressThread
+from sdcm.utils.replication_strategy_utils import NetworkTopologyReplicationStrategy
+
+warnings.filterwarnings(action="ignore", message="unclosed", category=ResourceWarning)
+
+
+class TestAddNewDc(LongevityTest):
+    """Test for procedure:
+     https://docs.scylladb.com/operating-scylla/procedures/cluster-management/add-dc-to-existing-dc/
+     """
+
+    def test_add_new_dc(self) -> None:  # pylint: disable=too-many-locals
+
+        assert self.params.get('n_db_nodes').endswith(" 0"), "n_db_nodes must be a list and last dc must equal 0"
+        system_keyspaces = ["system_auth", "system_distributed", "system_traces"]
+        self.reconfigure_keyspaces_to_use_network_topology_strategy(
+            keyspaces=system_keyspaces,
+            replication_factors={"eu-westscylla_node_west": 3}
+        )
+        self.prewrite_db_with_data()
+        read_thread, write_thread = self.start_stress_during_adding_new_dc()
+        # no need to change network topology
+        new_node = self.add_node_in_new_dc()
+
+        self.querying_new_node_should_return_no_data(new_node)  # verify issue #8354
+
+        self.reconfigure_keyspaces_to_use_network_topology_strategy(
+            keyspaces=system_keyspaces + ["keyspace1"],
+            replication_factors={"eu-westscylla_node_west": 3, "eu-west-2scylla_node_west": 1}
+        )
+
+        self.log.info("Running rebuild on each node in new DC")
+        new_node.run_nodetool(sub_cmd="rebuild -- eu-westscylla_node_west")
+
+        self.log.info("Running repair on all nodes")
+        for node in self.db_cluster.nodes:
+            node.run_nodetool(sub_cmd="repair -pr", publish_event=True)
+
+        # wait for stress to complete
+        self.verify_stress_thread(cs_thread_pool=read_thread)
+        self.verify_stress_thread(cs_thread_pool=write_thread)
+
+        self.verify_data_can_be_read_from_new_dc(new_node)
+        self.log.info("Test completed.")
+
+    def reconfigure_keyspaces_to_use_network_topology_strategy(self, keyspaces: List[str], replication_factors: dict[str, int]) -> None:
+        node = self.db_cluster.nodes[0]
+        self.log.info("Reconfiguring keyspace Replication Strategy")
+        network_topology_strategy = NetworkTopologyReplicationStrategy(
+            **replication_factors)
+        for keyspace in keyspaces:
+            cql = f"ALTER KEYSPACE {keyspace} WITH replication = {network_topology_strategy}"
+            node.run_cqlsh(cql)
+        self.log.info("Replication Strategies for {} reconfigured".format(keyspaces))
```

Comment:
recommended to use lazy interpolation:
`self.log.info("Replication Strategies for %s reconfigured", keyspaces)`

Reply:
removed, possibly it is not needed (left after investigation).